src/Workspaces/Core/Portable/Workspace/Host/Metadata/IAnalyzerAssemblyLoaderProvider.cs

```diff
@@ -2,9 +2,49 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.IO;
+using Microsoft.CodeAnalysis.Host.Mef;
+
 namespace Microsoft.CodeAnalysis.Host;
 
 internal interface IAnalyzerAssemblyLoaderProvider : IWorkspaceService
 {
     IAnalyzerAssemblyLoaderInternal GetShadowCopyLoader();
 }
+
+/// <summary>
+/// Abstract implementation of an analyzer assembly loader that can be used by VS/VSCode to provide a <see
+/// cref="IAnalyzerAssemblyLoader"/> with an appropriate path.
+/// </summary>
+internal abstract class AbstractAnalyzerAssemblyLoaderProvider : IAnalyzerAssemblyLoaderProvider
+{
+    private readonly Lazy<IAnalyzerAssemblyLoaderInternal> _shadowCopyLoader;
+
+    public AbstractAnalyzerAssemblyLoaderProvider(ImmutableArray<IAnalyzerAssemblyResolver> externalResolvers)
+    {
+        // We use a lazy here in case creating the loader requires MEF imports in the derived constructor.
+        _shadowCopyLoader = new(() => CreateShadowCopyLoader(externalResolvers));
+    }
+
+    public IAnalyzerAssemblyLoaderInternal GetShadowCopyLoader()
+        => _shadowCopyLoader.Value;
+
+    protected virtual IAnalyzerAssemblyLoaderInternal CreateShadowCopyLoader(ImmutableArray<IAnalyzerAssemblyResolver> externalResolvers)
+        => DefaultAnalyzerAssemblyLoader.CreateNonLockingLoader(
+            GetDefaultShadowCopyPath(),
+            externalResolvers: externalResolvers);
+
+    public static string GetDefaultShadowCopyPath()
+        => Path.Combine(Path.GetTempPath(), "Roslyn", "AnalyzerAssemblyLoader");
+}
```

Comment:
this logic changed from before.  We used to mix in the workspace kind into the path we wuold use.  But as @jasonmalinowski pointed out, we don't need to do that.  THe original idea was to keep the processes using separate directories so they wuold not stop on each other.  However, the ShadowCcopyLoader these all create doesn't need that.  First, each SCL puts its data under a unique guid root underneath this path.

Second, while each SCL tries to clean up anythnig left behind from prior sessions, it does so first by trying to get mutexes for the directories, and only deletes them if it can't.  So there isn't contention with other processes with their mutexes.

Reply:
I'm a little confused by this comment. _workspaceKind is still factoring into the path passed into WrapLoader, no?